### PR TITLE
Expose getConnectionString API 

### PR DESCRIPTION
**Pull request by VasuBhog**
To be used by the sql-database-projects extension to be used for the Add SQL Binding scenarios. 

In order to get the connectionString for SQL Server, we must set up a way to utilize the STS request for [GetConnectionStringRequest](https://github.com/Microsoft/sqltoolsservice/blob/main/src/Microsoft.SqlTools.ServiceLayer/Connection/ConnectionService.cs#L953). I utilized the connect() API in sql-db extension to first get the connectionUri and then pass that into the request param for it to then return the connection string.

### CODE REVIEW

File: src/models/contracts/connection.ts [hunk 1]
```diff
@@ -254,3 +254,26 @@ export class ListDatabasesResult {
...
+export namespace GetConnectionStringRequest {
+    export const type = new RequestType<GetConnectionStringParams, GetConnectionStringResult, void, void>('connection/getconnectionstring');
+}
+
+// Get Connection String request format
+export class GetConnectionStringParams {
+    // Connection key to lookup connection string for the provided connection Uri
+    public ownerUri: string;
+
+    // Indicates whether to include the password in the connection string
```
Comment (Charles-Gagnon) [inline src/models/contracts/connection.ts:270]: If you're adding comments like this they should be in js doc format (/** ... */)

File: src/controllers/connectionManager.ts [hunk 2]
```diff
@@ -259,6 +259,19 @@ export default class ConnectionManager {
         return (fileUri in this._connections && this._connections[fileUri].connecting);
     }
 
+    /**
+     * Get the connection string for the provided connection ID
+     * @param connectionId The connection ID for the connection.
+     * @param includePassword if password should be included in connection string.
+     * @returns connection string for the connection
+     */
+    public async getConnectionString(connectionUri: string, includePassword: boolean = false): Promise<string> {
```
Comment (Charles-Gagnon) [inline src/controllers/connectionManager.ts:268]: You take in "includePassword" but never actually set it

File: typings/vscode-mssql.d.ts [hunk 3]
```diff
@@ -76,6 +76,10 @@ declare module 'vscode-mssql' {
          */
         getDatabaseNameFromTreeNode(node: ITreeNodeInfo): string;
 
+        /**
+         * Get the connection string for the provided connection Uri
```
Comment (Charles-Gagnon) [inline typings/vscode-mssql.d.ts:80]: Missing parameters in js doc

**Review by Charles-Gagnon — CHANGES_REQUESTED** [review]

File: src/models/contracts/connection.ts (hunk 1 above)
Comment (VasuBhog) [reply src/models/contracts/connection.ts:270]: Sure thing will update this request section. It seems other requests in this file that I was following are not like that.

File: typings/vscode-mssql.d.ts [hunk 4]
```diff
@@ -76,6 +76,13 @@ declare module 'vscode-mssql' {
...
+         * @param connectionUri The URI of the connection to get the connection string for.
+         * @param includePassword to include password with connection string
+         * @returns connection string
+         */
+        getConnectionString(connectionUri: String, includePassword: boolean): Promise<string>;
```
Comment (Charles-Gagnon) [inline typings/vscode-mssql.d.ts:85]: includePassword should be optional per the params typing (and update the comment to mention default is false)

**Review by Charles-Gagnon — APPROVED** [review]

**Review by kisantia — APPROVED** [review]

File: src/controllers/connectionManager.ts [hunk 5]
```diff
@@ -259,6 +259,20 @@ export default class ConnectionManager {
         return (fileUri in this._connections && this._connections[fileUri].connecting);
     }
 
+    /**
+     * Get the connection string for the provided connection ID
+     * @param connectionId The connection ID for the connection.
```
Comment (chlafreniere) [inline src/controllers/connectionManager.ts:264]: connectionUri

**Pull request merged**